src/Microsoft.Framework.Configuration.Binder/ConfigurationBinder.cs

```diff
@@ -68,9 +68,9 @@ private static void BindProperty(PropertyInfo property, object propertyOwner, IC
             }
         }
 
-        private static object BindType(Type type, object typeInstance, IConfiguration configuration)
+        private static object BindType(Type type, object typeInstance, IConfigurationSection configuration)
         {
-            var configValue = configuration.Get(null);
+            var configValue = configuration.Value;
```

Comment:
@kirthik, @haok, @lodejard I have been thinking more about the restriction on ConfigurationBinder to operate only with non-root nodes and it seems it is wrong. 
From reading this code, what seems to be the case is that BindType will be called recursively at different levels and should just use `.Value` to obtain a scalar value for a scalar property. 
If we say `IConfigurationRoot` cannot have a value then we should just assume null for that case and only check for `.Value` if it is an `IConfigurationSection`. But we should not restrict the configuration passed. It should just be an IConfiguration.